### ISSUE TITLE
enum-glob-use causes `is defined multiple times` when identifier is shadowed

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#15185]
Using the following flags

`--force-warn clippy::enum-glob-use`

this code:

```rust
use self::Constructor::*;
struct IntRange {
    range: (),
}
enum Constructor {
    IntRange(IntRange),
}
fn main() {
    match Constructor::IntRange(IntRange { range: () }) {
        IntRange(x) => {}
    //  ^^^^^^^^^^^ wrong highlight, type of `x` not inferred
        Constructor::IntRange(x) => {}
    }
}
```

caused the following diagnostics:

```
    Checking _snippet_0 v0.1.0 (/tmp/icemaker_global_tempdir.IbV1PR0WiGNQ/icemaker_clippyfix_tempdir.7kuWY03r4mWr/_snippet_0)
warning: usage of wildcard import for enum variants
 --> src/main.rs:1:5
  |
1 | use self::Constructor::*;
  |     ^^^^^^^^^^^^^^^^^^^^ help: try: `self::Constructor::IntRange`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_glob_use
  = note: requested on the command line with `--force-warn clippy::enum-glob-use`

warning: `_snippet_0` (bin "_snippet_0") generated 1 warning (run `cargo clippy --fix --bin "_snippet_0"` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.12s

```

However after applying these diagnostics, the resulting code:

```rust
use self::Constructor::IntRange;
struct IntRange {
    range: (),
}
enum Constructor {
    IntRange(IntRange),
}
fn main() {
    match Constructor::IntRange(IntRange { range: () }) {
        IntRange(x) => {}
    //  ^^^^^^^^^^^ wrong highlight, type of `x` not inferred
        Constructor::IntRange(x) => {}
    }
}
```
no longer compiled:

```
    Checking _snippet_0 v0.1.0 (/tmp/icemaker_global_tempdir.IbV1PR0WiGNQ/icemaker_clippyfix_tempdir.7kuWY03r4mWr/_snippet_0)
error[E0255]: the name `IntRange` is defined multiple times
 --> src/main.rs:2:1
  |
1 | use self::Constructor::IntRange;
  |     --------------------------- previous import of the type `IntRange` here
2 | struct IntRange {
  | ^^^^^^^^^^^^^^^ `IntRange` redefined here
  |
  = note: `IntRange` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
  |
1 | use self::Constructor::IntRange as OtherIntRange;
  |                                 ++++++++++++++++

For more information about this error, try `rustc --explain E0255`.
error: could not compile `_snippet_0` (bin "_snippet_0" test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `_snippet_0` (bin "_snippet_0") due to 1 previous error

```

Version:
```
rustc 1.90.0-nightly (ed2d75978 2025-06-29)
binary: rustc
commit-hash: ed2d759783dc9de134bbb3f01085b1e6dbf539f3
commit-date: 2025-06-29
host: x86_64-unknown-linux-gnu
release: 1.90.0-nightly
LLVM version: 20.1.7

```
